### PR TITLE
Wave

### PR DESCRIPTION
**Pull request by meg-1**
### Describe your change:

**this algorithm returns a so called 'wave' of a given string [ a list of each of it's capitalized charachters one by one. ]**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by poyea** [comment]
Looks like this is a combination of #5883, and #6052.
I suggest you split them into individual PRs. That would be easier to review - unless i

**Comment by meg-1** [comment]
> Looks like this is a combination of #5883, and #6052. I suggest you split them into individual PRs. That would be easier to review

thank you! i closed the previous pull request [ #5883 ] just now so that it won't hinder the proccess of reviewing.

**Comment by ye290329** [comment]
> ### Describe your change:
> this algorithm returns a so called 'wave' of a given string [ a list of each of it's capitalized charachters one by one. ]
> 
> * [x]  Add an algorithm?
> * [ ]  Fix a bug or typo in an existing algorithm?
> * [ ]  Documentation change?
> 
> ### Checklist:
> * [x]  I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
> * [x]  This pull request is all my own work -- I have not plagiarized.
> * [x]  I know that pull requests will not be merged if they fail the automated tests.
> * [x]  This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
> * [x]  All new Python files are placed inside an existing directory.
> * [x]  All filenames are in all lowercase characters with no spaces or dashes.
> * [x]  All functions and variable names follow Python naming conventions.
> * [x]  All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
> * [x]  All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
> * [x]  All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
> * [x]  If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

close  the  repaper

**Comment by ye290329** [comment]
leave  this attention

**Comment by meg-1** [comment]
@Kush1101 could you please review this pull request??

**Comment by meg-1** [comment]
@Kush1101, could you please review this pull request??

**Comment by meg-1** [comment]
@poyea , could you please take a quick glance at my code, and advise how to avoid the 'tests are failing' label? so far while running locally the tests go thorugh quite easily, but here my code doesen't go through. thank you in advance!

**Comment by meg-1** [comment]
@Kush1101 could you please check out this pull request?

File: strings/wave.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    ['Cat', 'cAt', 'caT']
+    >>> wave('one')
+    ['One', 'oNe', 'onE']
+    >>> wave('book')
+    ['Book', 'bOok', 'boOk', 'booK']
+    """
+
+    """this algorithm returns a so called 'wave' of
+    a given string [ a list of each of it's capitalized
+    characters one by one??? ]"""
```
Comment (poyea) [inline strings/wave.py:13]:
```suggestion
    """
    Returns a so called 'wave' of a given string
    >>> wave('cat')
    ['Cat', 'cAt', 'caT']
    >>> wave('one')
    ['One', 'oNe', 'onE']
    >>> wave('book')
    ['Book', 'bOok', 'boOk', 'booK']
    """
```

Comment (poyea) [reply strings/wave.py:13]: We can merge these 2 blocks

Comment (meg-1) [reply strings/wave.py]: > We can merge these 2 blocks

thank you for the suggestion and review! just implemented it!

**Review by poyea — APPROVED** [review]
Thank you for your pull request!🤩

**Pull request merged**